### PR TITLE
Remove old, unused `hash_strength`

### LINKED ISSUE #6346: [Config] Remove hash_strength
`hash_strength` doesn't do anything anymore.

### PR DESCRIPTION
**Pull request by bobdenotter**
Fixes #6346

### CODE REVIEW

**Comment by GawainLynch** [comment]
Just one this. While it is completely correct according to the request & discussion in #6346, further discussion in #development on Slack was that in 3.4 we'd only use PasswordLib for legacy PHPass validation (at that point, an updated has is saved in the user's record), and use `password_hash()` and `password_verify()` … as when we implemented PasswordLib as a bridge, we were minimum PHP 5.4 and the other polyfills hadn't appeared, or been on our radar.

Basically looking to do something like this in 3.4:
```
$cycles = $app['access_control.hash.strength'];
$hash = password_hash($password, PASSWORD_DEFAULT, ['cost' => $cycles]);

$userEntity->setPassword($hash);
$repo->save($userEntity);
```

… as @rossriley put it 
> 8 might be ok for someone on shared hosting but I'd much rather use 12 or 13 on beefier hardware

**Comment by bobdenotter** [comment]
As far as I could tell, `access_control.hash.strength` is not used in the current codebase. 

And doesn't this take care of re-hashing the old hashed passwords? 

```
        // Rehash password if not using Blowfish algorithm
        if (!Blowfish::detect($userAuth->getPassword())) {
            $userEntity->setPassword($this->passwordFactory->createHash($password, '$2y$'));
            try {
                $this->getRepositoryUsers()->update($userEntity);
            } catch (NotNullConstraintViolationException $e) {
                // Database needs updating
            }
        }
```

So, new ones (plus old ones that are recognized, but then converted) are hashed using Crypt, which uses enough rounds. What am i missing?

**Comment by GawainLynch** [comment]
> So, new ones (plus old ones that are recognized, but then converted) are hashed using Crypt, 

Yes, but the line
```php
         if (!Blowfish::detect($userAuth->getPassword())) {
```

… is the only one we want to use PasswordLib for in v3.4+, and as we discussed the other day, removing it completely from v4.

> which uses enough rounds. What am i missing?

That "enough" is subjective, and people like @rossriley (and historically myself too), wind the cycles up on hardware that can cope with it, conversely if you're on a machine that really struggles, lower it … "you" obviously get both broken bits at that point :wink:

**Comment by bobdenotter** [comment]
> … is the only one we want to use PasswordLib for in v3.4+, and as we discussed the other day, removing it completely from v4.

Correct, but it doesn't use the configured number of rounds in any way. 

>  @rossriley (and historically myself too), wind the cycles up on hardware that can cope with it

`passwordFactory->createHash` doesn't seem to allow for a strength / rounds / cost parameter. `password_hash` does, but we don't use that yet. Are you saying that instead of removing, we should keep it around for future (re)use?

**Comment by GawainLynch** [comment]
> Correct, but it doesn't use the configured number of rounds in any way.

Exactly, which is why the conversation turned around in Slack. When we implemented PasswordLib, our minimum PHP requirement was 5.4 and we didn't have polyfills. Now we require PHP 5.5.9+ that hass all the needed bits.


Hence my original comment:

> Basically looking to do something like this in 3.4:
```
$cycles = $app['access_control.hash.strength'];
$hash = password_hash($password, PASSWORD_DEFAULT, ['cost' => $cycles]);

$userEntity->setPassword($hash);
$repo->save($userEntity);
```

**Comment by bobdenotter** [comment]
I'm sorry, but i don't understand where you're going with this? should we keep it around? Should I change the PR?

**Comment by GawainLynch** [comment]
OK, communication failure day … :disappointed: 

I am saying:
  * The commit *currently* in this PR is removing what we decided to keep, so needs to be dropped
  * Anything in core that is doing password hashing should be moved to use `password_hash($password, PASSWORD_DEFAULT, ['cost' => $cycles])` (where `$cycles = $app['access_control.hash.strength']`)
  * That can be done as part of an update and branch move for this PR (hence my failed attempt to explain an approach)
  * I'm right here for questions
  * We're at the 95% point of having all this done I think :smile:

**Comment by bobdenotter** [comment]
Ok, understood. I'll close this PR, and #6346 as well.
